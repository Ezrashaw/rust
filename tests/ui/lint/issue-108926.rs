#![forbid(unsafe_code)]

trait Foo {
    unsafe fn dangerous();
    //~^ ERROR declaration of an `unsafe` method [unsafe_obligation_define]
}

struct ImplOk;
impl Foo for ImplOk {
    #[forbid(unsafe_op_in_unsafe_fn)]
    unsafe fn dangerous() {}
    //~^ ERROR implementation of an `unsafe` method [unsafe_obligation_define]
}

struct ImplBad;
impl Foo for ImplBad {
    unsafe fn dangerous() {}
    //~^ ERROR implementation of an `unsafe` method [unsafe_obligation_discharge]
}

fn main() {}
