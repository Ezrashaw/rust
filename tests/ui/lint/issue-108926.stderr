error: declaration of an `unsafe` method
  --> $DIR/issue-108926.rs:4:5
   |
LL |     unsafe fn dangerous();
   |     ^^^^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> $DIR/issue-108926.rs:1:11
   |
LL | #![forbid(unsafe_code)]
   |           ^^^^^^^^^^^
   = note: `#[forbid(unsafe_obligation_define)]` implied by `#[forbid(unsafe_code)]`

error: implementation of an `unsafe` method
  --> $DIR/issue-108926.rs:11:5
   |
LL |     unsafe fn dangerous() {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

error: implementation of an `unsafe` method
  --> $DIR/issue-108926.rs:17:5
   |
LL |     unsafe fn dangerous() {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[forbid(unsafe_obligation_discharge)]` implied by `#[forbid(unsafe_code)]`

error: aborting due to 3 previous errors

